common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -854,7 +854,7 @@ def user_details_force_sync(auth_entry, strategy, details, user=None, *args, **k
     This step is controlled by the `sync_learner_profile_data` flag on the provider's configuration.
     """
     current_provider = provider.Registry.get_from_pipeline({'backend': strategy.request.backend.name, 'kwargs': kwargs})
-    if user and current_provider.sync_learner_profile_data:
+    if user and current_provider and current_provider.sync_learner_profile_data:
```

Comment:
Is there a relationship between this change and the site_id changes? Was `current_provider` guaranteed to exist before, but no longer, or is this independent? I'm wondering how careful we need to be about finding any other occurrences.

Reply:
It has been some time now but from what I remember, it was not guaranteed to exist even before this change. I'll see if I can try to reproduce the issue.

Reply:
In most cases the result is already checked for None value and then used. On the other hand, there are few references where it is assumed to return a value. I am not so sure about them being intentional or not, so I added the check for the ones which I can make sense of.